Add `Useful links` link promo (#2236)
BBC-archive/psammead: packages/components/psammead-useful-links/README.md

## Installation

```jsx
npm install @bbc/psammead-useful-links
```

## Props

| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |
| service | string | yes | N/A | `'news'` |
| script | object | yes | latin | { canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, } |
| usefulItems | array | yes | N/A | ['Mitocinmu da sauko da sautin labarai', 'Labaran BBC Hausa a text', 'Abokan huldar BBC Hausa'] |

## Usage

<!-- Description of the component usage -->

```jsx
import { latin } from '@bbc/gel-foundations/scripts';
import {
  UsefulLink,
  UsefulLinksLi,
  UsefulLinksUl,
  UsefulLinkItem,
} from './index';

const SingleUsefulItem = () => (
    <UsefulLink script={latin} service="news" usefulItems={["Labaran BBC Hausa a text"]} />
);

> const MultipleUsefulItems = () => (

Reviewer: Do you mind updating these examples to reflect the latest changes?

Author: Done:)

    <UsefulLinksUl role="list" 
                   usefulItems={['Mitocinmu da sauko da sautin labarai',
                    'Labaran BBC Hausa a text',
                    'Abokan huldar BBC Hausa']}>
        {usefulItems.map(item => {
            return (
                <UsefulLinksLi role="listitem">
                    <UsefulLinkItem script={latin} service="news">
                      {item}
                    </UsefulLinkItem>
                </UsefulLinksLi>
            );
        })}
    </UsefulLinksUl>
);
```

### When to use this component

This component can be used at any point on the page. `UsefulLinks` link promo should contain items that have a "PRO" `assetTypeCode` and "Guide" `contentType`.

### Accessibility notes

> The "Useful links" label is unique to the page. In case there are multiple useful items, they are announced by a screen reader as list items within a list. When there is only one useful item, a screen reader does not announce it as a list item. While navigating through the component using keybord only, pressing the TAB key changes state on focus.

Reviewer: I believe that we don't really need to specify the TAB navigation in our notes, as this is very specific to us, and this component could be used by anyone. @greenc05 Could you confirm?

```suggestion
In case there are multiple useful items, they are announced by a screen reader as list items within a list. When there is only one useful item, a screen reader does not announce it as a list item.

We have added the role `list` and `listitem` to the corresponding list items due to a VoiceOver bug to reinstate the list semantics.
```

## ⛔️ This is an alpha component ⛔️

This component is currently tagged as alpha and is not suitable for production use. Following the passing of an accessibility review this component will be marked as ready for production and the alpha tag removed.

## Description

The `UsefulLinks` component implements a Link Promo type that accesses links with information about various things to do with the world service website. `UsefulLinks` consists of a wrapper and a link element when there is only one useful item. In case there are multiple items, `UsefulLinks` consists of an unordered list with list items.

## Installation

```jsx
npm install @bbc/psammead-useful-links
```

## Props

| Argument  | Type | Required | Default | Example |

> | --------- | ---- | -------- | ------- | ------- |

Reviewer 2: Since you are exporting `UsefulLinkItem, UsefulLinkUl and UsefulLinksLi` they should both have their props listed on the documentation. Since they don't accept the same props as `UsefulLink`

| service | string | yes | N/A | `'news'` |

> | script | object | yes | latin | { canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, } |

Reviewer 3: Putting these examples into \`\` marks would make them easier to read as code.

> | usefulItems | array | yes | N/A | ['Mitocinmu da sauko da sautin labarai', 'Labaran BBC Hausa a text', 'Abokan huldar BBC Hausa'] |

Reviewer 2: There is no component using this as a prop.

| url | array | yes | N/A | ['https://www.bbc.com/igbo/afirika-49883577', 'https://www.bbc.com/igbo/afirika-49872694'] |

## Usage

<!-- Description of the component usage -->

```jsx
import { latin } from '@bbc/gel-foundations/scripts';
import {
  UsefulLink,
  UsefulLinksLi,
  UsefulLinksUl,
  UsefulLinkItem,
} from './index';

const SingleUsefulItem = () => (
    <UsefulLink script={latin} service="news" url={url}>
        {usefulItems[0]}
    </UsefulLink>
);

const MultipleUsefulItems = () => (
    <UsefulLinksUl>

>         {usefulItems.map((item, index) => {

Reviewer 2: ```jsx
{usefulItems.map((item) => {
            return (
                <UsefulLinksLi key={item.id}>
                    <UsefulLinkItem script={latin} service="news" href={item.url}>
                      {item.name}
                    </UsefulLinkItem>
                </UsefulLinksLi>
            );
        })}
```

            return (
                <UsefulLinksLi>
                    <UsefulLinkItem script={latin} service="news" href={url[index]}>
                      {item}
                    </UsefulLinkItem>
                </UsefulLinksLi>
            );
        })}
    </UsefulLinksUl>
);
```

### When to use this component

> This component can be used at any point on the page. `UsefulLinks` link promo should contain items that have a "PRO" `assetTypeCode` and "Guide" `contentType`.

Reviewer 2: ```suggestion
This component can be used at any point on the page.
```
This `PRO` `assetTypeCode` and `Guide` `contentType` is specific to simorgh. Since `psammead` is open source  it might be confusing to someone what those are.

### Accessibility notes

> In case there are multiple useful items, they are announced by a screen reader as list items within a list. When there is only one useful item, a screen reader does not announce it as a list item.

Reviewer 4: "**When** there are multiple useful items, they are announced by a screen reader as list items within a list. When there is only one useful item, a screen reader does not announce it as a list item.

> # psammead-useful-links - [![Known Vulnerabilities](https://snyk.io/test/github/bbc/psammead/badge.svg?targetFile=packages%2Fcomponents%2Fpsammead-useful-links%2Fpackage.json)](https://snyk.io/test/github/bbc/psammead?targetFile=packages%2Fcomponents%2Fpsammead-useful-links%2Fpackage.json) [![Dependency Status](https://david-dm.org/bbc/psammead.svg?path=packages/components/psammead-useful-links)](https://david-dm.org/bbc/psammead?path=packages/components/psammead-useful-links) [![peerDependencies Status](https://david-dm.org/bbc/psammead/peer-status.svg?path=packages/components/psammead-useful-links)](https://david-dm.org/bbc/psammead?path=packages/components/psammead-useful-links&type=peer) [![Storybook](https://raw.githubusercontent.com/storybooks/brand/master/badge/badge-storybook.svg?sanitize=true)](https://bbc.github.io/psammead/?path=/story/useful-links--containing-image) [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-useful-links.svg)](https://www.npmjs.com/package/@bbc/psammead-useful-links) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

Reviewer 2: `(https://bbc.github.io/psammead/?path=/story/useful-links--containing-image)` will be a dead link because there is no `useful-links` story with the title `containing-image` this should be `--default`

> ## ⛔️ This is an alpha component ⛔️

Reviewer: Do you mind moving this to the top of the README?

Author: Done

This component is currently tagged as alpha and is not suitable for production use. Following the passing of an accessibility review this component will be marked as ready for production and the alpha tag removed.

## Description

The `UsefulLinks` component implements a Link Promo type that accesses links with information about various things to do with the world service website. `UsefulLinks` consists of a wrapper and a link element when there is only one useful item. In case there are multiple items, `UsefulLinks` consists of an unordered list with list items.

## Installation

```jsx
npm install @bbc/psammead-useful-links
```

## UsefulLink Props

| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |
| service | string | yes | N/A | `'news'` |

> | script | object | yes | latin | `{ canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, }` |

Reviewer 2: ```suggestion
| script | object | yes | N/A | { canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, } |
```
Removed the marks around the example. It causes the storybook doc to have a horizontal scroll.

| href | string | yes | N/A | `'https://www.bbc.com/igbo/afirika-49883577'` |

## UsefulLinksUl Props

| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |

> | children | node | yes | N/A | `<UsefulLinksLi><UsefulLink service='news' script={latin href='https://www.bbc.com/igbo/afirika-49883577'> Mitocinmu da sauko da sautin labarai </UsefulLink></UsefulLinksLi>` |

Reviewer: ```suggestion
| children | node | yes | N/A | `<UsefulLinksLi><UsefulLink service='news' script={latin} href='https://www.bbc.com/igbo/afirika-49883577'> Mitocinmu da sauko da sautin labarai </UsefulLink></UsefulLinksLi>` |
```

## UsefulLinksLi Props

| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |

> | children | node | yes | N/A | `<UsefulLink service='news' script={latin href='https://www.bbc.com/igbo/afirika-49883577'> Mitocinmu da sauko da sautin labarai </UsefulLink>` |

Reviewer: ```suggestion
| children | node | yes | N/A | `<UsefulLink service='news' script={latin} href='https://www.bbc.com/igbo/afirika-49883577'> Mitocinmu da sauko da sautin labarai </UsefulLink>` |
```
